lib/cdo/aws/resource_management/ec2_instance_for_deletion.rb

```diff
@@ -0,0 +1,67 @@
+# EC2InstanceForDeletion class controls the deletion process of an EC2 instance and provides methods to evaluate
+# if an instance is eligible for deletion.
+#
+# It includes functionalities to check if the API termination is disabled, if the instance has an associated
+# stack for deletion, if the stack is critical, and if the instance has been stopped for a minimum duration.
+#
+# Additionally, it retrieves necessary information such as the stack name and stop time, allowing users to
+# determine the eligibility of an EC2 instance for deletion based on defined criteria.
+class EC2InstanceForDeletion
+  # Constants defining tags and critical stacks
+  DISABLED_API_TERMINATION_TAG = 'disableApiTermination'.freeze
+  CLOUD_FORMATION_TAG = 'aws:cloudformation:stack-name'.freeze
+  CRITICAL_STACKS = %w[autoscale-prod test staging levelbuilder adhoc-bugcrowd]
+
+  # Initialize EC2 instance and optional deletion configuration options
+  def initialize(instance, opts = {})
+    @instance = instance
+    @deletion_tag = opts[:deletion_tag]
+    @minimum_time_stopped_in_seconds = opts.fetch(:minimum_time_stopped_in_seconds, 6 * 24 * 60 * 60) # Default 6 days
+    @_dry_run = opts[:dry_run]
```

Comment:
Is this used anywhere?

Reply:
not yet

That's why I added an _

I'll be using it for delete and tagging methods that will be added on top of this